### PR TITLE
Userlist

### PR DESCRIPTION
**Pull request by ejohnson44**
Added in User objects and completely redid UserList to match

### CODE REVIEW

File: fanClub/src/com/lastcrusade/fanclub/CoreActivity.java [hunk 1]
```diff
@@ -52,7 +53,10 @@ public void onCreate(Bundle savedInstanceState){
 
         //Add user to user list
         CustomApp curApp = ((CustomApp)getApplication());
-        curApp.getUserList().addUser(BluetoothUtils.getLocalBluetoothName());
+        
+        //TODO: Move this to something like connect activity or the connection fragment
```
Comment (dgreenhalgh) [inline fanClub/src/com/lastcrusade/fanclub/CoreActivity.java:57]: When are we planning on doing this?

File: fanClub/src/com/lastcrusade/fanclub/model/SongMetadata.java [hunk 2]
```diff
@@ -10,18 +10,18 @@
 
     private String album;
 
-    private String username; // Bluetooth id of person who added song to
+    private String macAddress; // macAddress of person who added song to
                              // playlist
```
Comment (dgreenhalgh) [inline fanClub/src/com/lastcrusade/fanclub/model/SongMetadata.java]: u. g. l. y.

File: fanClub/src/com/lastcrusade/fanclub/model/UserColors.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
+package com.lastcrusade.fanclub.model;
+
+public class UserColors {
+    //color set not finished yet - might change some or all colors
+    private ColorEntry[] colors = new ColorEntry[] {
+        
+        new ColorEntry(0xFF02AD8C),
```
Comment (dgreenhalgh) [inline fanClub/src/com/lastcrusade/fanclub/model/UserColors.java:7]: Could we, as a rule, put a comment anywhere there's a hardcoded color declaring what the color is?

File: fanClub/src/com/lastcrusade/fanclub/components/PlaylistFragment.java [hunk 4]
```diff
@@ -27,23 +30,22 @@
...
 
-        //TODO: get the userlist better
-        final CustomApp curApp = (CustomApp) this.getActivity().getApplication();
 
         playlistServiceServiceLocator.setOnBindListener(new ServiceLocator.IOnBindListener() {
             @Override
             public void onServiceBound() {
                 metadataList = getPlaylistService().getPlaylist();
-                setListAdapter(new MusicListAdapter(PlaylistFragment.this.getActivity(), metadataList.getList(), curApp.getUserList().getUsers()));
+                setListAdapter(new MusicListAdapter(PlaylistFragment.this.getActivity(), metadataList.getList(), curApp.getUserList()));
```
Comment (bryansills) [inline fanClub/src/com/lastcrusade/fanclub/components/PlaylistFragment.java:48]: :8ball: I like that we created an empty playlist at the beginning and once we get connected to the service we populate the list. i probably wouldn't have thought of that :moneybag: good code. :cactus:

File: fanClub/src/com/lastcrusade/fanclub/model/UserList.java [hunk 5]
```diff
@@ -1,65 +1,92 @@
...
-import java.util.Hashtable;
-import java.util.Iterator;
 import java.util.List;
-import java.util.Map.Entry;
 
+import com.lastcrusade.fanclub.util.BluetoothUtils;
+
+
+/*
+ * TODO: determine if we want to keep track of an actual
```
Comment (dgreenhalgh) [inline fanClub/src/com/lastcrusade/fanclub/model/UserList.java]: I think that this falls outside of the realm of our minimum requirements, but it's definitely something we can talk about in the future

File: fanClub/src/com/lastcrusade/fanclub/model/UserList.java [hunk 6]
```diff
@@ -1,65 +1,92 @@
...
-        nextColor = 5 % colors.length;
+        userColors = new UserColors();
+        connectedUsers = new ArrayList<User>();
     }
 
-    public UserList(String user){
-        connectedUsers.put(user, colors[nextColor]);
-        nextColor = (nextColor + 1) % colors.length;
+    public UserList(String bluetoothID, String macAddress){
+        this();
```
Comment (bryansills) [inline fanClub/src/com/lastcrusade/fanclub/model/UserList.java:17]: what does this() do?

File: fanClub/src/com/lastcrusade/fanclub/model/UserList.java [hunk 7]
```diff
@@ -1,65 +1,92 @@
...
+    public void addUser(String bluetoothID, String macAddress){
+        //check to make sure that the user isn't already in the list before adding
+        if(getUserByMACAddress(macAddress)==null){
+            connectedUsers.add(new User(bluetoothID, macAddress, userColors.getNextAvailableColor()));
+        }
     }
 
-    public void removeUser(String user){
-        connectedUsers.remove(user);
+    //untested
```
Comment (dgreenhalgh) [inline fanClub/src/com/lastcrusade/fanclub/model/UserList.java:28]: Let's make an item in Trello to make sure we don't forget to test this :hamster:

File: fanClub/src/com/lastcrusade/fanclub/util/UserListAdapter.java [hunk 8]
```diff
@@ -11,38 +8,37 @@
...
-    private Hashtable<String, String> users;
+    private UserList users;
     
-    public UserListAdapter(Context mContext, String[] usernames, Hashtable<String,String> users){
-        this.usernames = usernames;
+    public UserListAdapter(Context mContext, UserList users){
         this.users = users;
         this.mContext = mContext;
     }
     @Override
```
Comment (dgreenhalgh) [inline fanClub/src/com/lastcrusade/fanclub/util/UserListAdapter.java:24]: newline

File: fanClub/src/com/lastcrusade/fanclub/model/UserList.java (hunk 6 above)
Comment (ejohnson44) [inline fanClub/src/com/lastcrusade/fanclub/model/UserList.java:17]: constructor chaining - it calls the other constructor that  does the list instantiation stuff so we don't have to have the exact two same lines of code twice

**Comment by dgreenhalgh** [comment]
Approved by @bryansills and I. The only outstanding issue is the addition of comments specifying each color, but that will be added in the near future.

File: fanClub/src/com/lastcrusade/fanclub/model/UserColors.java (hunk 3 above)
Comment (reidbaker) [inline fanClub/src/com/lastcrusade/fanclub/model/UserColors.java:7]: I see we didn't do this. Is the play to do it later or what? @ejohnson44

File: fanClub/src/com/lastcrusade/fanclub/model/UserList.java (hunk 6 above)
Comment (reidbaker) [inline fanClub/src/com/lastcrusade/fanclub/model/UserList.java:17]: Very clever! I need to use this in my pull request.

File: fanClub/src/com/lastcrusade/fanclub/model/UserColors.java (hunk 3 above)
Comment (dgreenhalgh) [inline fanClub/src/com/lastcrusade/fanclub/model/UserColors.java:7]: I made a note of it farther down the pull request, but the plan is to incorporate it into @ejohnson44's next pull request

**Pull request merged**